File: 01_arrays/6_stock_buy_and_sell.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        if n == 1:
            return 0
        result = 0
        curr_mini = prices[0]
        for item in prices[1:]:
            curr_mini = min(curr_mini, item)
            result = max(result, item - curr_mini)
        return result

File: 01_arrays/6_stock_buy_and_sell.rs
use std::cmp::{max, min};

impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut mini: i32 = i32::MAX;
        let mut res: i32 = 0;
        for price in prices {
            mini = min(mini, price);
            res = max(res, price - mini);
        }
        res
    }
}
